Fix Babelfish log function incorrect return result

RIC06X opened the pull request:
### Description

Previously Babelfish natural log function mapped to postgres base 10 log function and returned incorrect result. In addition,  Babelfish's two argument log function had base as first argument and value as second argument which is different from TSQL. This commit remapped Babelfish log() function to postgres ln() function and fixed the incorrect log value, and fixed the argument ordering issue for the two argument log function.

### Issues Resolved

BABEL-730

### Test Scenarios Covered ###
* return type test for log(x), log(x, y), log10(x)
* return value test for log(x), log(x, y), log10(x)
* function/procedure usage test for log(x), log(x, y), log10(x)
* view usage test for log(x), log(x, y), log10(x)
* implicit casting (string value) test for log(x), log(x, y), log10(x)



### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

RIC06X committed:
[BABEL-730] Fix bbf log function produce incorrect result

Previously babelfish natural log function maps to pg base 10 log
function and produced incorrect result. This commit remapped babelfish
log function mapping and fixed the incorrect log value issue.

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
[BABEL-730] Fix LOG Reserved Keyword issue

Added LOG10 keyword in ANTLR and added log and log10 keyword in
pg delimited keyword list

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

forestkeeper commented on test/JDBC/input/BABEL-730-vu-cleanup.sql:
```diff
@@ -0,0 +1,39 @@
+use test_babel730
+go
```
We should add this to mvu schedule files

RIC06X replied:
Solved

RIC06X committed:
[BABEL-730] Add tests to upgrade schedule files

Added tests to upgrade schedule files

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
[BABEL-730] Remove views in upgrade test

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

xhfanhe commented on contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.4.0--3.5.0.sql:
```diff
@@ -37,6 +37,17 @@ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION sys.bbf_log10(IN arg1 FLOAT)
+RETURNS FLOAT  AS 'babelfishpg_tsql','numeric_log10' LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION sys.bbf_log10(FLOAT) TO PUBLIC;
```
Is public access necessary?

RIC06X replied:
no, it is not necessary to have public access, I'll remove the grant to public stmt.

xhfanhe commented on test/JDBC/expected/BABEL-730-vu-cleanup.out:
```diff
@@ -0,0 +1,21 @@
+-- drop table/view/function/procedure
+drop table test_log
```
Where is this table created?

RIC06X replied:
this table is created in verify.sql, it is created by 'select into' stmt

xhfanhe commented on test/python/expected/upgrade_validation/expected_dependency.out:
```diff
@@ -224,6 +224,9 @@ Function sys.bbf_binary_varbinary_cmp(sys.bbf_binary,sys.bbf_varbinary)
 Function sys.bbf_get_context_info()
 Function sys.bbf_get_current_physical_schema_name(text)
 Function sys.bbf_is_shared_schema(text)
+Function sys.bbf_log(double precision)
```
Please add upgrade tests for bbf_log functions as well (if it's public accessible)

RIC06X replied:
Tests for bbf_log functions are at the end of verify.sql

RIC06X committed:
[BABEL-730] Remove public access to bbf_log

Removed public access to bbf_log and bbf_log10

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
[BABEL-730] Add view test and bbf_log dependency test

Added additional test set for upgrade test before 15.6 or 16.1.
Added view test and bbf_log dependency test for upgrade tests that on
latest version.

Task: BABEL-730
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>
